extensions/terminal-suggest/src/tokens.ts

```diff
@@ -0,0 +1,33 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { TerminalShellType } from './terminalSuggestMain';
+
+export const enum TokenType {
+	Command,
+	Argument,
+}
+
+const shellTypeResetChars: { [key: number]: string[] | undefined } = {
+	[TerminalShellType.Bash]: ['>', '>>', '<', '2>', '2>>', '&>', '&>>', '|', '|&', '&&', '||', '&', ';', '(', '{', '<<'],
+	[TerminalShellType.Zsh]: ['>', '>>', '<', '2>', '2>>', '&>', '&>>', '<>', '|', '|&', '&&', '||', '&', ';', '(', '{', '<<', '<<<', '<('],
+	[TerminalShellType.PowerShell]: ['>', '>>', '<', '2>', '2>>', '*>', '*>>', '|', '-and', '-or', '-not', '!', '&', '-eq', '-ne', '-gt', '-lt', '-ge', '-le', '-like', '-notlike', '-match', '-notmatch', '-contains', '-notcontains', '-in', '-notin']
+};
+
+const defaultShellTypeResetChars = shellTypeResetChars[TerminalShellType.Bash]!;
+
+export function getTokenType(ctx: { commandLine: string; cursorPosition: number }, shellType: TerminalShellType | undefined): TokenType {
+	const spaceIndex = ctx.commandLine.substring(0, ctx.cursorPosition).lastIndexOf(' ');
+	if (spaceIndex === -1) {
+		return TokenType.Command;
+	}
+	const tokenIndex = spaceIndex === -1 ? 0 : spaceIndex + 1;
+	const previousTokens = ctx.commandLine.substring(0, tokenIndex).trim();
```

Comment:
rename `priorCommandFragment`?

Reply:
Was trying to lean into the "token" wording here since chances are as we improve this we'll turn closer and closer into a real lexer and understand more of the tokens.